javascript/javascript2/week2/main.js

```diff
@@ -0,0 +1,98 @@
+        const liElement = document.createElement("li");
+        liElement.classList.add("card");
+        ulElement.appendChild(liElement);
+
+        const imgElement = document.createElement("img");
+        imgElement.src = "https://picsum.photos/200";
+        liElement.appendChild(imgElement);
+
+        // create a h1 element and append to the li, assign the product name to its text
+        const h2Element = document.createElement("h2");
+        h2Element.innerHTML = product.name;
+        liElement.appendChild(h2Element);
+
+        // create a span element and append to the li, assign the product price to its text
+        const divElement = document.createElement("div");
+        divElement.innerHTML = `Price: ${product.price}`;
+        liElement.appendChild(divElement);
+
+        // create another span element and append to the li, assign the product rating to its text
+        const spanElement2 = document.createElement("span");
+        spanElement2.innerHTML = `Rating: ${product.rating}`;
+        liElement.appendChild(spanElement2);
+
+        // create another span element and append to the li, assign the heart icon to its text
+        const buttonElement = document.createElement("button");
+        buttonElement.innerHTML = `&#9825;`;
+        buttonElement.classList.add("add-item");
+        liElement.appendChild(buttonElement);
+        liElement.id = h2Element.innerHTML.slice(0, 1) + product.rating;
+        // add event listener after clicking the like button
+        const likeButtons = document.getElementsByClassName("add-item");
+        const likeButtonsArray = Array.from(likeButtons);
+        likeButtonsArray.forEach((button) => button.addEventListener("click", addProduct));
+    }
+}
+
+renderProducts(products);
+// Get all the user input values and save
+const inputElement1 = document.getElementById("search_products_name");
+const inputElement2 = document.getElementById("search_min_price");
+const inputElement3 = document.getElementById("search_max_price");
+const inputElement4 = document.getElementById("search_products_rating");
+const filterButton = document.getElementById("filter_button");
+// Set some initial values; for prices, if user does not specify any values, take the lowest and highest prices as min and maxPrice
+const productsPrice = products.map((product) => product.price);
+let minPrice = productsPrice.reduce((accumulator, currentValue) => (accumulator < currentValue) ? accumulator : currentValue, products[0].price);
+let maxPrice = productsPrice.reduce((accumulator, currentValue) => (accumulator > currentValue) ? accumulator : currentValue, products[0].price);
+let minRating = 0;
+
+function onNameChange(event) {
+    renderProducts(products.filter((product) => product.name.toLowerCase().startsWith(event.target.value.toLowerCase())));
+}
+// After clicking on filter button, set the predefined variebles as per user choice and call render method
+function onFilter() {
```

Comment:
It's hard to figure out what this function should do from the name. Try give it a bit more descriptive name

Reply:
Yes, will change that. Thank you for the feedback.